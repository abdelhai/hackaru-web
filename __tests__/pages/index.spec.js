import { Store } from 'vuex-mock-store';
import Factory from '@/__tests__/__setups__/factory';
import Index from '@/pages/index';

describe('Index', () => {
  let wrapper;
  let factory;

  const $store = new Store({
    getters: {
      'activities/getWorkingActivities': () => [
        {
          id: 1,
          description: 'Review',
          startedAt: '2019-01-01T01:23:45'
        }
      ]
    }
  });

  beforeEach(() => {
    factory = new Factory(Index, {
      mocks: {
        $store
      }
    });
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  it('dispatch activities/getWorkingActivities', () => {
    factory.shallow();
    expect($store.dispatch).toHaveBeenCalledWith(
      'activities/getWorkingActivities'
    );
  });

  describe('when click add button', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.find('.add-button').vm.$emit('click');
    });

    it('show new activity modal', () => {
      expect(factory.options.mocks.$modal.show).toHaveBeenCalledWith(
        'activity'
      );
    });
  });
});
