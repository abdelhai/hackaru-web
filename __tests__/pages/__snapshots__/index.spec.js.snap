// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index render correctly 1`] = `
<section>
  <content-header-stub>
    <heading-stub>
      title
    </heading-stub>
     
    <btn-stub
      aria-label="ariaLabels.add"
      class="is-primary is-circle has-dropshadow add-button"
      type="button"
    >
      <icon-stub
        name="plus-icon"
      />
    </btn-stub>
  </content-header-stub>
    
  <p
    class="empty"
  >
    
    empty
  
  </p>
</section>
`;
